dispatch/dispatch.go

```diff
@@ -255,11 +254,13 @@ func (d *Dispatcher) Groups(routeFilter func(*Route) bool, alertFilter func(*typ
 
 // Stop the dispatcher.
 func (d *Dispatcher) Stop() {
+	d.mtx.Lock()
```

Comment:
Ah the failure is legit! Alertmanager crashes on startup because the `Stop()` function is called with a `nil` receiver. We need this:

```go
       if d == nil {
           return
       }
	d.mtx.Lock()
	if d.cancel == nil {
            d.mtx.Unlock()
	    return nil
	}
        ...
```